lib/mayaUsd/resources/ae/usdschemabase/ae_template.py

```diff
@@ -660,6 +659,123 @@ def get(self):
+        self.useMetadata = hasattr(ufe.SceneItem, "getGroupMetadata") and hasattr(ufe.Value, "typeName")
+
+    @property
+    def GROUP(self):
+        # When using meta data we use a special group name which will automatically
+        # write to the session layer.
+        return 'SessionLayer-Autodesk' if self.useMetadata else 'Autodesk'
+
+    @property
+    def USE_OUTLINER_COLOR(self):
+        # To be more generic and not Maya specific we write out the custom data
+        # as "Text Color" instead of "Outliner Color".
+        key = 'Use Text Color'
+        return key if self.useMetadata else (self.GROUP + ':' + key)
+
+    @property
+    def OUTLINER_COLOR(self):
+        key = 'Text Color'
+        return key if self.useMetadata else (self.GROUP + ':' + key)
+
+    def onCreate(self, *args):
+        '''Add two fields for outliner color to match Maya objects.'''
+        l1 = mel.eval('uiRes(\"m_AEdagNodeCommon.kUseOutlinerColor\");')
+        l2 = mel.eval('uiRes(\"m_AEdagNodeCommon.kOutlinerColor\");')
+        self.useOutlinerColor = cmds.checkBoxGrp(label1=l1, ncb=1,
+                                                 ann=getMayaUsdLibString('kUseOutlinerColorAnn'),
+                                                 cc1=self._onUseOutlinerColorChanged)
+        self.outlinerColor = cmds.colorSliderGrp(label=l2,
+                                                 ann=getMayaUsdLibString('kOutlinerColorAnn'),
+                                                 cc=self._onOutlinerColorChanged)
+
+        # Update all the custom data controls.
+        self.refresh()
+
+    def onReplace(self, *args):
+        pass
+
+    def clear(self):
+        cmds.checkBoxGrp(self.useOutlinerColor, edit=True, v1=False)
+        cmds.colorSliderGrp(self.outlinerColor, edit=True, rgb=(0,0,0))
+
+    def refresh(self):
+        try:
+            if self.useMetadata:
+                # Use Ufe SceneItem metadata to get values.
+                useOutlinerColor = self.item.getGroupMetadata(self.GROUP, self.USE_OUTLINER_COLOR)
+                assert(not useOutlinerColor.empty())
+                outlinerColor = self.item.getGroupMetadata(self.GROUP, self.OUTLINER_COLOR)
+                cmds.checkBoxGrp(self.useOutlinerColor, edit=True, v1=bool(useOutlinerColor))
+                if not outlinerColor.empty() and (outlinerColor.typeName() == "ufe.Vector3d"):
+                    # Color is stored as double3 USD custom data.
+                    clr = ufe.Vector3d(outlinerColor)
+                    cmds.colorSliderGrp(self.outlinerColor, edit=True,
+                                        rgb=(clr.x(), clr.y(), clr.z()))
+            else:
+                # Get the custom data directly from USD.
+                useOutlinerColor = self.prim.GetCustomDataByKey(self.USE_OUTLINER_COLOR)
+                if useOutlinerColor:
+                    cmds.checkBoxGrp(self.useOutlinerColor, edit=True, v1=useOutlinerColor)
```

Comment:
Are we only updating the checkbox when the value is True? why?

Reply:
I had intended this to be a check if the value returned from GetCustomDataByKey is not None. I think I need to be explicit in that check here (and below).

Reply:
Good catch, I reworked the code a bit to properly test the return values for each case.